File: LeetCode/191_number_of_1_bits.py
#!/usr/bin/env python3
# pylint: disable=invalid-name, too-few-public-methods, no-self-use,missing-function-docstring, unused-wildcart-import
'''
    LeetCode solution class
'''
from typing import List, Optional


class Solution:
    '''
    LeetCode solution class
    '''

    def hammingWeight(self, n: int) -> int:
        hamming_weight = 0
        for p in range(0, 32):
            if n & pow(2, p) > 0:
                hamming_weight += 1
        return hamming_weight


solution = Solution()
print(solution.hammingWeight(0b00000000000000000000000000001011))  # 3
print(solution.hammingWeight(0b00000000000000000000000010000000))  # 1
print(solution.hammingWeight(0b11111111111111111111111111111101))  # 31

File: LeetCode/191_number_of_1_bits.php
<?php

declare(strict_types=1);

use Solution as GlobalSolution;

/**
 * Template Soultion class for LeetCode problems
 * 
 */
class Solution
{
    /**
     * @var int $hw
     */
    private int $hw = 0;

    /**
     * @param int $n
     * @return int
     */
    public function hammingWeight(int $n): int
    {
        $this->hw = 0;
        for ($i = 0; $i < 32; $i++) {
            if ($n & pow(2, $i)) {
                $this->hw++;
            }
        }
        return $this->hw;
    }
}
// test suite
$solution = new GlobalSolution();
printf("%d\n", $solution->hammingWeight(0b00000000000000000000000000001011)); // 3
printf("%d\n", $solution->hammingWeight(0b00000000000000000000000010000000)); // 1
printf("%d\n", $solution->hammingWeight(0b11111111111111111111111111111101)); // 31
